controllers/applicationvolumereplication_controller.go

```diff
@@ -120,106 +126,263 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
+
+	// Skip this subscription if a manifestwork already exist
+	if found, err := r.findVRGManifestWork(avr, subscription); err != nil {
+		return nil, requeue
+	} else if found {
+		r.Log.Info("Skipping existing mainifestwork for subscription", "name", subscription.Name)
+
+		return nil, !requeue
+	}
+
+	// This subscription is ready for manifest (VRG) creation
+	r.Log.Info("Subscription is unpaused", "Name", subscription.Name)
+
+	placementDecision, err := r.processUnpausedSubscription(subscription)
+	if err != nil {
+		return nil, requeue
+	}
+
+	return &placementDecision, !requeue
+}
+
+func (r *ApplicationVolumeReplicationReconciler) isSubsriptionPausedForDR(labels map[string]string) bool {
+	return labels != nil &&
+		labels[RamenDRLabelName] != "" &&
+		strings.EqualFold(labels[RamenDRLabelName], "protected") &&
+		labels[subv1.LabelSubscriptionPause] != "" &&
+		strings.EqualFold(labels[subv1.LabelSubscriptionPause], "true")
+}
+
+// processPausedSubscription selects the target cluster from the subscription or
+// from the user selected cluster and restores all PVs (belonging to the subscription)
+// to the target cluster and then it unpauses the subscription
+func (r *ApplicationVolumeReplicationReconciler) processPausedSubscription(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) error {
+	r.Log.Info("Processing paused subscription", "name", subscription.Name)
+	// find target cluster (which can be the failover cluster)
+	homeClusterName := r.findNextHomeCluster(avr, subscription)
+
+	if homeClusterName == "" {
+		return errorswrapper.New("failed to find new home cluster")
+	}
+
+	err := r.deleteExistingManfiestWork(avr, subscription)
+	if err != nil {
+		return err
+	}
+
+	err = r.restoreBackedupPVs(subscription, homeClusterName)
+	if err != nil {
+		return err
+	}
+
+	return r.unpauseSubscription(subscription)
+}
+
+func (r *ApplicationVolumeReplicationReconciler) findVRGManifestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) (bool, error) {
+	const notFound = false
+
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
+		mw := &ocmworkv1.ManifestWork{}
+		vrgMWName := fmt.Sprintf(ManifestWorkNameFormat, subscription.Name, subscription.Namespace, "vrg")
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: vrgMWName, Namespace: d.HomeCluster}, mw)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				r.Log.Info("ManifestWork not found for Subscription", "name", subscription.Name)
+
+				return notFound, nil
+			}
+
+			return notFound, errorswrapper.Wrap(err, "failed to retrieve manifestwork")
+		}
+
+		return found, nil
+	}
+
+	return notFound, nil
+}
+
+func (r *ApplicationVolumeReplicationReconciler) deleteExistingManfiestWork(
+	avr *ramendrv1alpha1.ApplicationVolumeReplication,
+	subscription *subv1.Subscription) error {
+	if d, found := avr.Status.Decisions[subscription.Name]; found {
```

Comment:
(future) We should find and update/delete VRG MW from all clusters other than the failoverCluster as of now, instead of looking at the status in the future?